output_delegation.go

```diff
@@ -164,7 +158,7 @@ func (d *DelegationOutput) UnlockConditionSet() UnlockConditionSet {
 }
 
 func (d *DelegationOutput) ImmutableFeatureSet() FeatureSet {
```

Comment:
In my opinion this function should be removed from the delegation output.
You should create a new interface for chained outputs with the immutable feature.

```diff

// ChainOutput is a type of Output which represents a chain of state transitions.
type ChainOutput interface {
	Output
	// Chain returns the ChainID to which this Output belongs to.
	Chain() ChainID
-	// ImmutableFeatureSet returns the immutable FeatureSet this output contains.
-	ImmutableFeatureSet() FeatureSet
}

+// ChainOutputImmutable is a type of Output which represents a chain of state transitions with immutable features.
+type ChainOutputImmutable interface {
+	ChainOutput
+	// ImmutableFeatureSet returns the immutable FeatureSet this output contains.
+	ImmutableFeatureSet() FeatureSet
+}
```

Then you can remove the `return vm.IsIssuerOnOutputUnlocked(current, vmParams.WorkingSet.UnlockedIdents)` check in `delegationGenesisValid`.